(MVP): Add landscape support for the PIN layout.

manpreeeeeet opened the pull request:
MVP: Add support for landscape mode for the PIN layout.

![image](https://github.com/user-attachments/assets/9024af2f-bf4b-4630-969c-7bea243c7fcb)

manpreeeeeet committed:
(MVP): Add landscape support for the PIN layout.

manpreeeeeet commented on app/src/main/java/com/xinto/mauth/ui/component/pinboard/PinBoard.kt:
```diff
@@ -29,54 +31,140 @@ fun PinBoard(
+            val rowCount = 4
+            val rowWidth = 3
+            val columnCount = 3
+            val columnButtons = buildList {
```
Currently I am transforming the rows into columns so we can use the FlowColumn + weight to make our buttons adjust the height using `weight(1f)`. 

Another option is generating a new `PinBoard` state with the buttons depending on the orientation.

X1nto replied:
Don't forget to use `remember {}` when dealing with data. Otherwise that `buildList` will get executed on every rerender. I think `remember(columnCount, rowCount) {}` should be fine, but those are constants anyway, so it's better to inline

manpreeeeeet commented on app/src/main/java/com/xinto/mauth/ui/component/pinboard/PinBoard.kt:
```diff
@@ -29,54 +31,140 @@ fun PinBoard(
-                        onLongClick =
+            }
+            columnButtons.forEach { button ->
+                when (button) {
```
We cannot extract this common code into a `Composable` since it uses modifiers like `weight` which is an extension function that changes based on `FlowRow` or `FlowColumn` scope.

manpreeeeeet committed:
Manually calculate the width required to
fit 3 pin buttons.

In landscape mode, height is the limiting constraint. So, this calculation tries to
fit exactly 3 buttons in a row using their
`PinButtonDefaults.PinButtonMinSize`.

manpreeeeeet committed:
Add small button size and update
PinSetupScreen and PinRemoveScreen to use
the small button size.

manpreeeeeet committed:
Address feedback: Merge paddings.
